plugins/search-react/src/components/SearchResultGroup/SearchResultGroup.stories.tsx

```diff
@@ -0,0 +1,209 @@
+  id: 'storybook.test-route',
+});
+
+const searchApiMock = new MockSearchApi({
+  results: [
+    {
+      type: 'techdocs',
+      document: {
+        location: 'search/search-result1',
+        title: 'Search Result 1',
+        text: 'Some text from the search result 1',
+      },
+    },
+    {
+      type: 'techdocs',
+      document: {
+        location: 'search/search-result2',
+        title: 'Search Result 2',
+        text: 'Some text from the search result 2',
+      },
+    },
+  ],
+});
+
+export default {
+  title: 'Plugins/Search/SearchResultGroup',
+  component: SearchResultGroup,
+  decorators: [
+    (Story: ComponentType<{}>) =>
+      wrapInTestApp(
+        <TestApiProvider apis={[[searchApiRef, searchApiMock]]}>
+          <Grid container direction="row">
+            <Grid item xs={12}>
+              <Story />
+            </Grid>
+          </Grid>
+        </TestApiProvider>,
+        { mountedRoutes: { '/hello': routeRef } },
+      ),
+  ],
+};
+
+export const Default = () => {
+  const [query] = useState<Partial<SearchQuery>>({
+    types: ['techdocs'],
+  });
+
+  return (
+    <SearchResultGroup
+      icon={<DocsIcon />}
+      title="Documentation"
+      query={query}
+    />
+  );
+};
+
+export const WithNoResults = () => {
+  const [query] = useState<Partial<SearchQuery>>({
+    types: ['techdocs'],
+  });
+
+  return (
+    <TestApiProvider apis={[[searchApiRef, new MockSearchApi()]]}>
+      <SearchResultGroup
+        icon={<DocsIcon />}
+        title="Documentation"
+        query={query}
+        noResultItemsText="No results found"
+      />
+    </TestApiProvider>
+  );
+};
+
+export const WithFilters = () => {
+  const [query, setQuery] = useState<Partial<SearchQuery>>({
+    types: ['software-catalog'],
+  });
+
+  const filterOptions = [
+    {
+      label: 'Lifecycle',
+      value: 'lifecycle',
+    },
+    {
+      label: 'Owner',
+      value: 'owner',
+    },
+  ];
+
+  const handleFilterAdd = useCallback(
+    (key: string) => () => {
+      setQuery(prevQuery => {
+        const { filters: prevFilters, ...rest } = prevQuery;
+        const newFilters = { ...prevFilters, [key]: undefined };
+        return { ...rest, filters: newFilters };
+      });
+    },
+    [],
+  );
+
+  const handleFilterChange = useCallback(
+    (key: string) => (value: JsonValue) => {
+      setQuery(prevQuery => {
+        const { filters: prevFilters, ...rest } = prevQuery;
+        const newFilters = { ...prevFilters, [key]: value };
+        return { ...rest, filters: newFilters };
+      });
+    },
+    [],
+  );
+
+  const handleFilterDelete = useCallback(
+    (key: string) => () => {
+      setQuery(prevQuery => {
+        const { filters: prevFilters, ...rest } = prevQuery;
+        const newFilters = { ...prevFilters };
+        delete newFilters[key];
+        return { ...rest, filters: newFilters };
+      });
+    },
+    [],
+  );
+
+  return (
+    <SearchResultGroup
```

Comment:
It might be valuable to give an example of how you can compose your SearchResult component with SearchResultGroups. Maybe here https://backstage.io/storybook/?path=/story/plugins-search-searchresult--default?

Reply:
Hmm.. I believe they cannot be conceptually composited because "SearchResults" depends on a context and "SearchResultsGroup" does not (create its own query), I think they are an alternative to each other, or you use "SearchResults" or "SearchResultsGroup".

Reply:
Hmmm, so let's discuss that a bit more. The use case I have is that when I go through the getting started docs https://backstage.io/docs/features/search/getting-started#adding-search-to-the-frontend and set up my search for the frontend, I create the search page, and instead of rendering a list of search results, I would like to compose my search page with result groups instead. How would I do that?

Reply:
Thanks for explaining, now I understand! That would be more templates for a search page, is there anywhere I can create templates for the search page in the storybook?

Reply:
I think we don't want examples of the search page necessarily, but of the SearchResults. 

Search Page -> its just the page container
Search Results -> its the container responsible for rendering results (no matter if the results are rendered in a list, in a table, or in groups) 

So I guess what we want to achieve is that you should be able to use SearchResults to render one or many ResultGroups too?  as its just one way of rendering your results?

> "SearchResults" depends on a context and "SearchResultsGroup" does not (create its own query)

does that need to be the case? can not the `SearchResultsGroup` take in the results as a prop? which the `SearchResults` already provide?

Maybe this very early frontend component architecture can help me explain https://github.com/backstage/backstage/issues/4445#issuecomment-826603847

Reply:
As we discussed, I added examples of list and group layouts 🙂